File: concepts/big-o/constant-time-example/java/ConstantTimeExample.java
public class ConstantTimeExample {
    public static void main(String[] args){
        int[] arr1 = {1, 2, 3, 4, 5};
        // 3
        System.out.println(getArrayElement(arr1, 2));
    }

    public static int getArrayElement(int[] arr, int index){
        return arr[index];
    }
}

File: concepts/big-o/constant-time-example/cpp/constantTimeExample.cpp
// g++ constantTimeExample.cpp -o constantTimeExample -std=c++20 && ./constantTimeExample

#include <iostream>
#include <array>

template <size_t N>
int getArrayElement(const std::array<int, N>& arr, size_t index);

int main()
{
    std::array<int, 5> arr1 = { 1, 2, 3, 4, 5 };

    // 3
    std::cout << getArrayElement(arr1, 2) << std::endl;

    return 0;
}

template <size_t N>
int getArrayElement(const std::array<int, N>& arr, size_t index)
{
    return arr[index];
}